php/libraries/Database.class.inc

```diff
@@ -144,10 +147,10 @@ class Database
      */
     function connect(
         $database,
-        $username,
-        $password,
-        $host,
-        $trackChanges = true
+        ?string $username,
+        ?string $password,
+        ?string $host,
```

Comment:
This one seems weird.. what does it mean to pass those as null?

Reply:
Honestly not totally sure. It requires further research. However when these are set to just strings (`null` not allowed) Travis throws up a ton of errors during the execution of unit tests. Something to the effect of "MockDatabase::connect() requires a string parameter for param X but null given".

I don't know enough about our automated testing to know what to do about that, so I allowed changed this function to allow for null. I'm very open to suggestions here.

Reply:
@kongtiaowang Any thoughts on how to solve that?

Reply:
@johnsaigle other than resolving this ^ issue and travis, i think everything else looks good. will test in a sec

Reply:
apache php7.2 upgrade not working for me so can't test at the moment

Reply:
IMO we should go ahead and merge this. I think the nullable strings thing is likely a quirk of PHPUnit. In any case if any of the params is null then the DB will error out anyway due to invalid credentials.